posthog/api/insight.py

```diff
@@ -738,6 +739,16 @@ def activity(self, request: request.Request, **kwargs):
         activity_page = load_activity(scope="Insight", team_id=self.team_id, item_id=item_id, limit=limit, page=page)
         return activity_page_response(activity_page, limit, page, request)
 
+    @action(methods=["POST"], detail=False)
+    def cancel(self, request: request.Request, **kwargs):
+        if "client_query_id" not in request.data:
+            raise serializers.ValidationError({"client_query_id": "Field is required."})
+        sync_execute(
```

Comment:
This won't work in a multi-node setup sadly - every query will go to a random node.

Reply:
Dang! Could we send the kill query to every node individually?

Reply:
as per slack, looks like ON CLUSTER should do the trick here. It's behind a feature flag so I'll get this merged and test it out for our team